src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -757,6 +838,12 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         AddAction(*cmd, keys);
     }
 
+    void ActionMap::DeleteUserCommand(const winrt::hstring& cmdID)
+    {
+        _ActionMap.erase(cmdID);

```

Comment:
```suggestion
        _ActionMap.erase(cmdID);
        _changeLog.emplace(IDKey)
```
This one I'm iffy on, so I want to discuss here.

Currently, there isn't a place where we're tracking if a user added an ID. In the JSON, the user would be using "id" in two cases for Action Map:
1. create a new action in the `"actions"` block (i.e. `{"command": "terminalChat", "id": "User.terminalChat"}`)
   - That's fine? We care about action arg usage and that's already being handled.
2. create a key binding in the `"keybindings"`block (i.e. `{"id": "User.terminalChat", "keys": "ctrl+?}`)
   - That's fine too. We care about "keys" usage and that's already being handled.

(related: `"id"` is used in the new tab menu, but we're just tracking `"action"` there which is sufficient)

Long-term, however, we want `_changeLog` to be a way for us to know if the user changed anything. So maybe we do want to apply the code suggestion and update case 1 to add `"id"` too? Thoughts?

Reply:
This one is tough because I imagine most users won't be actually adding their own IDs - it will just be autogenerated for them. So distinguishing between us generating an ID for them vs an ID they added or changed is... tricky